### PR TITLE
Support for editing alarms

### PR DESCRIPTION
**Pull request by languitar**
As requested in #301, this is a work in progress pull request for editing alarms.

### CODE REVIEW

File: khal/khalendar/event.py [hunk 1]
```diff
@@ -342,6 +342,40 @@ def summary(self):
...
+    def alarms(self):
+        """
+        Returns a list of all alarms in th original event that we can handle. Unknown types of
+        alarms are ignored.
+        """
+        return [(a.get('TRIGGER').dt, a.get('DESCRIPTION'))
+                for a in self._vevents[self.ref].subcomponents
+                if a.name == 'VALARM' and self._can_handle_alarm(a)]
+
+    def updated_alarms(self, alarms):
```
Comment (untitaker) [inline khal/khalendar/event.py]: You can use the `alarms.setter` decorator here, FWIW, but better to stay consistent.

But: please rename to `update_alarms`

Comment (languitar) [inline khal/khalendar/event.py]: Oops, sure. this was a typo.

**Comment by geier** [comment]
Thank you! As far as I see, the only thing that really is left, is validation of user inputs, right?

While it doesn't make the event editor any prettier, that's not your fault at all.
We probably should improve the theming a lot, e.g. by making titles and user input field better separable.

**Comment by languitar** [comment]
Yes, that's what is missing to my mind. And optionally supporting more types of alarms. It's funny that the Ical spec supports so many more types of alarms than ical on OSX supports. ;) Unfortunately I am currently a bit out of time for fixing this. I hope I find some time in a few days.

**Comment by sdx23** [comment]
I'm a big fan of this feature. Thanks for the good work, keep it going :)

**Comment by geier** [comment]
I created a new widget type `ValidateEdit`, this should make validation really easy, just pass along a validation function and you should be good to go. If you need any help, just say so. If you don't have the time for it, I will rebase and merge your branch if that's ok with you.

**Comment by languitar** [comment]
@geier great, I've seen the commit and already thought that this might be useful. I hope I will find some time at the weekend to finalize this.

**Comment by fpytloun** [comment]
Looks great! Only few more wishes:
- it would be good to setup default alarm for every new event in config (eg. notification 10 minute before event start)
- it would be great to introduce some `--alarm` command (to be executed from cron) which would execute action at alarm time (eg. I have [alarm.sh](https://gist.github.com/fpytloun/4acc181d67d78b47b23b) for this purpose)

**Comment by geier** [comment]
I agree with @fpytloun first suggestion (but I'll happily merge this without it), but I won't do anything about the second one (see #148). However, if someone else wants to implement this (and keep supporting it), I'm happy to accept a PR.

**Comment by geier** [comment]
Hi @languitar, I took the liberty of rebasing your PR and making the EditWidgets in the DurationWidget validate input (see branch `alarms`). If you are okay with that, I'll merge that branch. Could you add yourself to the `AUTHORS.txt` file, please?

**Comment by languitar** [comment]
@geier Thank you. I'm sorry, I didn't find the time to do this so far.

Should I overwrite my own branch with yours and add the authors change or what is the idea?

**Comment by geier** [comment]
@languitar nothing to be sorry for, you did all the work. I don't mind really, you cant just do a PR against my `alarms` branch, another PR with just the AUTHOR change against master, rebase your own `alarms` branch against current master and cherry-pick my commit (or just do it by hand, I don't care about atttribution). Just do what is most convenient for you.

**Comment by languitar** [comment]
Ok, I have updated my own branch.

I also changed the return statement for the `unsigned_integer` function as discussed.

File: khal/ui/widgets.py [hunk 2]
```diff
@@ -424,3 +424,107 @@ def keypress(self, size, key):
...
                 return
         return super().keypress(size, key)
+
+
+class DurationWidget(urwid.WidgetWrap):
+
+    @staticmethod
+    def unsigned_int(number):
+        """test if `number` can be converted to a positive int"""
+        try:
```
Comment (hobarrera) [inline khal/ui/widgets.py]: Looks like my previous comment got lost in the void during rebasing.

IMHO, this looks simpler/cleaner:

```python
try:
    return int(number) >= 0
except ValueError:
    return False
```

**Comment by geier** [comment]
Thank you for your patience and this big improvement!

**Pull request merged**